### PR TITLE
Refactoring authenticated XHR calls, allows basic auth

### PR DESCRIPTION
**Pull request by PanierAvide**
This PR was made mainly to allow using basic authentication against OSM API. Use case is usage of this library on server-side, for sending edits through a single OSM account. Related to current work around caresteouvert.fr, see https://github.com/osmontrouge/covid19_map/issues/13

### CODE REVIEW

File: src/helpers/xhr.js [hunk 1]
```diff
@@ -63,3 +63,73 @@ export function fetch(url, options = {}) {
...
+          );
+        }
+
+        return response;
+      })
+      .catch(error => {
+        throw new RequestException(error);
+      })
+      .then(response => response.text());
+  }
```
Comment (GuillaumeAmat) [inline src/helpers/xhr.js]: I would throw an exception in an `else` case, to let the developer know that it misses something.

**Comment by GuillaumeAmat** [comment]
Great to see that @PanierAvide 👌

**Comment by GuillaumeAmat** [comment]
Also, you decreased the coverage so the checks are failing

File: src/helpers/xhr.js (hunk 1 above)
Comment (PanierAvide) [reply src/helpers/xhr.js]: Thanks for your feedback, I added an else case.

**Comment by PanierAvide** [comment]
Added more tests for this xhr script ;-)

File: src/helpers/__tests__/xhr.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { fetch, authxhr } from '../xhr';
+
+describe('XHR helpers', () => {
+  describe('fetch', () => {
+    it('Gets data from given URL', async () => {
+      const res = await fetch(
+        'https://files.pavie.info/depot/remote/example.json'
```
Comment (GuillaumeAmat) [inline src/helpers/__tests__/xhr.test.js]: Basing tests on remote resources is not a good idea as you introduce uncontrolled side effects. You should only test your code only.

So you should create some stubs for `auth.xhr` and `crossFetch`. Jest has the right methods to do that.
The idea is to verify that `auth.xhr` and `crossFetch` are called, when it is right, and with the right arguments by using `authxhr`.

Comment (PanierAvide) [reply src/helpers/__tests__/xhr.test.js]: Sure, I changed to use conventions from Jest and proper mocks. Thanks for your feedback.

**Comment by PanierAvide** [comment]
So I updated tests according to your feedback, thank you for your time spent to review my edits :smiley:

**Comment by GuillaumeAmat** [comment]
Thanks for your changes!

The Travis CI report was stuck but the CI was green, I checked. So I merged the PR anyway.

**Pull request merged**